src/app/shared/object-collection/shared/badges/access-status-badge/access-status-badge.component.ts

```diff
@@ -57,5 +63,12 @@ export class AccessStatusBadgeComponent {
       map((status: string) => `access-status.${status.toLowerCase()}.listelement.badge`),
       catchError(() => observableOf('access-status.unknown.listelement.badge'))
     );
+
+    // stylesheet based on the access status value
+    this.accessStatus$.pipe(
+      map((accessStatusClass: string) => accessStatusClass.replace(/\./g, '-'))
+    ).subscribe((accessStatusClass: string) => {
+      this.accessStatusClass = accessStatusClass;
+    });
```

Comment:
Any time you add a subscription, please [store it in a component property](https://github.com/DSpace/dspace-angular/blob/8bf4ba812614e5ed34b2808b79d7e364fd01c2db/src/app/dso-shared/dso-edit-metadata/dso-edit-metadata.component.ts#L122) and then [unsubscribe from it, in the ngOnDestroy method of the component](https://github.com/DSpace/dspace-angular/blob/8bf4ba812614e5ed34b2808b79d7e364fd01c2db/src/app/dso-shared/dso-edit-metadata/dso-edit-metadata.component.ts#L255-L259). That means that we'll unsubscribe whenever that component stops being rendered.

If you don't do that these subscriptions will remain in memory indefinitely, and especially in a case like this, where a new subscription is opened for every search result, that could lead to problems

Reply:
I've added the onDestroy to deal with the subscription.